lib/security/WSSecurity.js

```diff
@@ -20,7 +21,17 @@ function WSSecurity(username, password, options) {
   }
 
   this._hasTimeStamp = options.hasTimeStamp || typeof options.hasTimeStamp === 'boolean' ? !!options.hasTimeStamp : true;
+  /*jshint eqnull:true */
+  if (options.hasNonce != null) {
```

Comment:
I use `options.hasNonce == null` to check if `hasNonce` is `undefined` or `null`. If we use `options.hasOwnProperty('hasNonce')`, it will return `true` when `hasNonce` is explicitly set to `undefined` or `null`. Either way is fine but it doesn't really matter for this case, IMO.